bot/exts/halloween/spooknamerate.py

```diff
@@ -0,0 +1,387 @@
+    # the bot is running
+    data = RedisCache()
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+
+        names_data = self.load_json(Path('bot', 'resources', 'halloween', 'spooknamerate_names.json'))
+        self.first_names = names_data['first_names']
+        self.last_names = names_data['last_names']
+        # the names are from https://www.mockaroo.com/
+
+        self.name = None
+
+        self.bot.loop.create_task(self.load_vars())
+
+        self.first_time = None
+        self.poll = False
+        self.announce_name.start()
+        self.checking_messages = asyncio.Lock()  # Define an asyncio.Lock() to make sure the dictionary isn't changed
+        # when checking the messages for duplicate emojis'
+
+    async def load_vars(self) -> None:
+        """Loads the variables that couldn't be loaded in __init__."""
+        self.first_time = await self.data.get('first_time', True)
+        self.name = await self.data.get('name')
+
+    @group(name="spooknamerate", invoke_without_command=True)
+    async def spook_name_rate(self, ctx: Context) -> None:
+        """Get help on the Spook Name Rate game."""
+        help_embed = Embed(
+            title="Spook Name Rate",
+            description=f"Help for the `{self.bot.command_prefix}spooknamerate` command",
+            color=Colours.soft_orange,
+        )
+
+        for field in HELP_MESSAGE:
+            help_embed.add_field(
+                name=field['title'],
+                value=field['val'],
+                inline=field['inline']
+            )
+
+        await ctx.send(embed=help_embed)
+
+    @spook_name_rate.command(name='list', aliases=["all", "entries"])
+    async def list_entries(self, ctx: Context) -> None:
+        """Send all the entries up till now in a single embed."""
+        await ctx.send(embed=await self.get_responses_list(final=False))
+
+    @spook_name_rate.command(name='name')
+    async def tell_name(self, ctx: Context) -> None:
+        """Tell the current random name (name)."""
+        if not self.poll:
+            await ctx.send(f"The name is **{self.name}**")
+            return
+
+        await ctx.send(
+            f"The name ~~is~~ was **{self.name}**. The poll has already started, so you cannot "
+            "add an entry."
+        )
+
+    @spook_name_rate.command(name="add", aliases=["register"])
+    async def add_name(self, ctx: Context, *, name: str) -> None:
+        """Use this command to add/register your spookified name."""
+        if self.poll:
+            logger.info(f"{ctx.message.author} tried to add a name, but the poll had already started.")
+            await ctx.send("Sorry, the poll has started! You can try and participate in the next round though!")
+            return
+
+        message = ctx.message  # get the message
+
+        for data in [json.loads(x[1]) for x in await self.messages.items()]:
```

Comment:
```suggestion
        for data in [json.loads(user_json) for _, user_json in await self.messages.items()]:
```
Let's use better names here